client/src/components/MapView.vue

```diff
@@ -223,11 +232,20 @@ export default defineComponent({
       }
     },
   },
+  mounted() {
+    // Show current data layer on init.
+    this.updateMapOnDataLayerChange(this.state.currentDataLayer, null);
+  },
   watch: {
+    // Listens to app state to toggle layers.
     'state.currentDataLayer': function(newDataLayer: DataLayer, oldDataLayer: DataLayer) {
       this.updateMapOnDataLayerChange(newDataLayer, oldDataLayer);
     },
-  },
+    // Listens to query param to toggle layers.
+    'routerLayer': function(newLayer: string) {
```

Comment:
does this not get triggered when you update URL in setDataLayerVisibility (/wouldn't that cause a loop)?

Reply:
Basically the only thing is  router.currentRoute.value.query.layer != styleLayerId in setDataLayerVisibility(). We need both the app state and the router url to update the data layer and both need to stay in style. So this one setter needs to update both.

Reply:
Gotcha, thanks!